janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CqlBridgeRecordReader.java

```diff
@@ -0,0 +1,377 @@
+                .build();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        if (cluster != null) {
+            session = cluster.connect(quote(keyspace));
+        }
+
+        if (session == null) {
+            throw new RuntimeException("Can't create connection session");
+        }
+
+        //get negotiated serialization protocol
+        nativeProtocolVersion = cluster.getConfiguration().getProtocolOptions().getProtocolVersion().toInt();
+
+        // If the user provides a CQL query then we will use it without validation
+        // otherwise we will fall back to building a query using the:
+        //   inputColumns
+        //   whereClauses
+        cqlQuery = CqlConfigHelper.getInputCql(conf);
+        // validate that the user hasn't tried to give us a custom query along with input columns
+        // and where clauses
+        if (StringUtils.isNotEmpty(cqlQuery) && (StringUtils.isNotEmpty(inputColumns) ||
+            StringUtils.isNotEmpty(userDefinedWhereClauses))) {
+            throw new AssertionError("Cannot define a custom query with input columns and / or where clauses");
+        }
+
+        if (StringUtils.isEmpty(cqlQuery)) {
+            cqlQuery = buildQuery();
+        }
+        log.trace("cqlQuery {}", cqlQuery);
+        distinctKeyIterator = new DistinctKeyIterator();
+        log.trace("created {}", distinctKeyIterator);
+    }
+
+    public void close() {
+        if (session != null) {
+            session.close();
+        }
+        if (cluster != null) {
+            cluster.close();
+        }
+    }
+
+    private static class KV {
+        private final StaticArrayBuffer key;
+        private ArrayList<Entry> entries;
+
+        KV(StaticArrayBuffer key) {
+            this.key = key;
+        }
+
+        void addEntries(Collection<Entry> toAdd) {
+            if (entries == null) {
+                entries = new ArrayList<>(toAdd.size());
+            }
+            entries.addAll(toAdd);
+        }
+    }
+
+    @Override
+    public StaticBuffer getCurrentKey() {
+        return currentKV.key;
+    }
+
+    @Override
+    public Iterable<Entry> getCurrentValue() throws IOException {
+        return currentKV.entries;
+    }
+
+    public float getProgress() {
+        if (!distinctKeyIterator.hasNext()) {
+            return 1.0F;
+        }
+
+        // the progress is likely to be reported slightly off the actual but close enough
+        float progress = ((float) distinctKeyIterator.totalRead / totalRowCount);
+        return progress > 1.0F ? 1.0F : progress;
+    }
+
+    public boolean nextKeyValue() throws IOException {
+        Map<StaticArrayBuffer, Map<StaticBuffer, StaticBuffer>> kv = distinctKeyIterator.next();
+        if (kv == null) {
+            return false;
+        } else {
+            int size = kv.size();
+            assert size == 1 : "There should be one entry in key value map, found: " + size;
```

Comment:
No, I do want a descriptive message and to avoid the inadvertent cost of `getField()`, I declared the variable which I much prefer to:
```
assert kv.size() == 1 : "There should be one entry in key value map, found: " + kv.size();
```